### PR TITLE
NCBI  OMIM disease as gene fix

### PR DESCRIPTION
**Pull request by TomConlin**
- addresses #961
- restore default taxon in more general form. 
   - (no recollection of why I commented them out months ago)
- write out the omim dicts for type & replacement status
- repair incidental test breakage  from category branch
- begin sorting and de-duping ntriples earlier rather than later

### CODE REVIEW

File: dipper/sources/NCBIGene.py [hunk 1]
```diff
@@ -452,13 +453,16 @@ def _add_gene_equivalencies(self, dbxrefs, gene_id, taxon):
...
-                    elif omim_num in self.omim_type and\
+                    if omim_num in self.omim_type and\
                             self.omim_type[omim_num] == self.globaltt['gene']:
                         model.addXref(gene_id, dbxref_curie)
                     else:
-                        continue  # no equivilance between ncbigene and omin-nongene
+                        # OMIM disease/phenotype is not considered a gene at all
+                        # no equivilance between ncbigene and omin-nongene
+                        # and ncbi is never a human clique leader in any case
+                        break
```
Comment (kshefchek) [inline dipper/sources/NCBIGene.py:465]: wondering if a break here makes sense, wouldn't this remove any xref after the omim id?

Comment (TomConlin) [reply dipper/sources/NCBIGene.py:465]: more specifically, any xref after an omim that does not resolve to a gene.  yes

Comment (kshefchek) [reply dipper/sources/NCBIGene.py]: i think these xrefs are still valid, for example for NCBIGene:55906, we previously xref'ed it to ENSEMBL:ENSG00000126970 and OMIM:300897, and made it equivalent to ENSEMBL:ENSG00000126970, OMIM:300897, OMIM:309605, HGNC:24931

On this branch theres only one xref to OMIM:300897 and one eq to OMIM:300897.

Maybe there are cases where a bad OMIM xref indicates the ncbi identifier is not a gene?  But in this case NCBIGene:55906 looks like a gene despite it.

I am in favor of fewer equivalent class triples, especially to ensembl as we tend to get better xrefs from other sources (HGNC, the mod dbs)

Comment (TomConlin) [reply dipper/sources/NCBIGene.py]: Sorry if I am being dense but what are you suggesting?

Comment (kshefchek) [reply dipper/sources/NCBIGene.py]: I think ENSEMBL:ENSG00000126970 should be an xref for NCBIGene:55906

Comment (TomConlin) [reply dipper/sources/NCBIGene.py]: No argument from me, but nothing in this (current) PR prevents it.

Comment (TomConlin) [reply dipper/sources/NCBIGene.py]: Wonder if it is our all screaming caps curie-prefix 'ENSEMBL'

Comment (TomConlin) [reply dipper/sources/NCBIGene.py]: maybe more of a modelling concern but 
isn't being a dbxref redundant considering it is already an equivalenct class which must have uri usable as a xref

Comment (kshefchek) [reply dipper/sources/NCBIGene.py]: looks good to me, I missed where you changed this (was looking at 0deb70)

Comment (TomConlin) [reply dipper/sources/NCBIGene.py]: the UI  presented a sub optimal where it continued to display the old, only flagging it with "updated" in washed out tones.

but all for the best I think as it got me to stare at it harder and fix/streamline that section of logic

**Pull request merged**